Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,332 @@
+      try
       {
-        var result = RebarFreeFormValidationResult.Success;
-        rebar ??= DB.Structure.Rebar.CreateFreeForm(Doc, barType, host, curveLoops, out result);
-        if (result != RebarFreeFormValidationResult.Success)
-        {
-          appObj.Update(status: ApplicationObject.State.Failed, logItem: "Freeform Rebar could not be created from closed curves.");
-          return appObj;
-        }
+        rebar = DB.Structure.Rebar.CreateFromCurves(
+          Doc,
+          barStyle,
+          barType,
+          startHook,
+          endHook,
+          CurrentHostElement,
+          normal,
+          curves,
+          startHookOrientation,
+          endHookOrientation,
+          true,
+          true
+        );
       }
-      else if (openCurves.Count > 0)
+      catch (Autodesk.Revit.Exceptions.ArgumentNullException nullEx)
       {
-        //rebar ??= DB.Structure.Rebar.CreateFromCurves(Doc, barType, host, curveLoops, out result);
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: nullEx.Message);
       }
-      else
+      catch (Autodesk.Revit.Exceptions.ArgumentOutOfRangeException rangeEx)
       {
-        appObj.Update(status: ApplicationObject.State.Failed, logItem: "No convertible rebar curves where found.");
-        return appObj;
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: rangeEx.Message);
+      }
+      catch (Autodesk.Revit.Exceptions.ArgumentsInconsistentException inconsistentEx)
+      {
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: inconsistentEx.Message);
       }
+      catch (Autodesk.Revit.Exceptions.ArgumentException argException)
+      {
+        var info =
+          "The element host is not a valid rebar host. -or- The input curves "
+          + "contains at least one curve which is not a bound Line or bound Arc and is not "
+          + "supported for this operation. -or- curves do not form a valid CurveLoop. -or- The "
+          + "input curves contains at least one helical curve and is not supported for this operation.";
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: $"{argException.Message}: {info}");
+      }
+      catch (Autodesk.Revit.Exceptions.DisabledDisciplineException disciplineException)
+      {
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: disciplineException.Message);
+      }
+      finally
+      {
+        if (rebar != null)
+        {
+          SetInstanceParameters(rebar, speckleRebar);
 
-      if (speckleRevitRebar != null)
-        SetInstanceParameters(rebar, speckleRevitRebar);
+          // set additional params
+          if (speckleRebar.barPositions > 0)
+          {
+            rebar.NumberOfBarPositions = speckleRebar.barPositions;
+          }
 
-      appObj.Update(status: ApplicationObject.State.Created, createdId: rebar.UniqueId, convertedItem: rebar);
+          // deleting instead of updating for now!
```

Comment:
do we only offer replace atm? How does that sit with UX for the user, will they expect this?

Reply:
We should generally try to update and existing element instead of replace when possible, but this is rather tricky for most revit elements. As a first pass, we've been allowing replace instead of a true update (you'll see this in some other to native conversions). But this is a good point, we haven't clarified `update` behavior across and within connectors.

Reply:
I think this is consistent then with our not being consistent with update behaviour and probably a much bigger problem. Maybe, as a team, we should consider whether update is part of the story in future and size appropriately.